features/mbedtls/targets/TARGET_CRYPTOCELL310/aes_alt.h

```diff
@@ -0,0 +1,111 @@
+  *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *  See the License for the specific language governing permissions and
+  *  limitations under the License.
+  *
+  */
+
+#ifndef __AES_ALT__
+#define __AES_ALT__
+
+#include "ssi_aes.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+struct sw_aes_context
+{
+    int nr;                     /*!<  number of rounds  */
+    uint32_t *rk;               /*!<  AES round keys    */
+    uint32_t buf[68];           /*!<  unaligned data    */
+    int sw_key_set;
+};
+
+typedef struct
+{
+    struct {
+        SaSiAesUserContext_t CC_Context;
+        SaSiAesEncryptMode_t CC_cipherFlag;
+        SaSiAesUserKeyData_t CC_KeyData;
+    } cc_aes_context;
+    struct sw_aes_context sw_context;
+}
+mbedtls_aes_context;
+
+void mbedtls_aes_init( mbedtls_aes_context *ctx );
+
+void mbedtls_aes_free( mbedtls_aes_context *ctx );
+
+int mbedtls_aes_setkey_enc( mbedtls_aes_context *ctx, const unsigned char *key,
+                    unsigned int keybits );
+
+int mbedtls_aes_setkey_dec( mbedtls_aes_context *ctx, const unsigned char *key,
+                    unsigned int keybits );
+
+int mbedtls_aes_crypt_ecb( mbedtls_aes_context *ctx,
+                    int mode,
+                    const unsigned char input[16],
+                    unsigned char output[16] );
+
+#if defined(MBEDTLS_CIPHER_MODE_CBC)
+int mbedtls_aes_crypt_cbc( mbedtls_aes_context *ctx,
+                    int mode,
+                    size_t length,
+                    unsigned char iv[16],
+                    const unsigned char *input,
+                    unsigned char *output );
+#endif /*MBEDTLS_CIPHER_MODE_CBC*/
+
+#if defined(MBEDTLS_CIPHER_MODE_CTR)
+int mbedtls_aes_crypt_ctr( mbedtls_aes_context *ctx,
+                       size_t length,
+                       size_t *nc_off,
+                       unsigned char nonce_counter[16],
+                       unsigned char stream_block[16],
+                       const unsigned char *input,
+                       unsigned char *output );
+#endif /*MBEDTLS_CIPHER_MODE_CTR*/
+
+#if defined(MBEDTLS_CIPHER_MODE_CFB)
+/*
+ * AES CFB is not Supported by Cryptocell.
+ * AES-CFB will use SW implementation
+ */
+
+int mbedtls_aes_crypt_cfb128( mbedtls_aes_context *ctx,
+                       int mode,
+                       size_t length,
+                       size_t *iv_off,
+                       unsigned char iv[16],
+                       const unsigned char *input,
+                       unsigned char *output );
+
+
+int mbedtls_aes_crypt_cfb8( mbedtls_aes_context *ctx,
+                    int mode,
+                    size_t length,
+                    unsigned char iv[16],
+                    const unsigned char *input,
+                    unsigned char *output );
+#endif /*MBEDTLS_CIPHER_MODE_CFB */
+
+
```

Comment:
Note that the software implementation transforms the encrypted data back to big endian:
https://github.com/nvlsianpu/mbed/pull/5/files#diff-c82c0749f19741dad8445c869cc48e4fR522
Are we sure that this alternative implementation outputs the same data (format-wise) as the software implementation?